sympy/matrices/expressions/hadamard.py

```diff
@@ -119,7 +119,18 @@ def shape(self):
         return self.base.shape
 
     def _entry(self, i, j, **kwargs):
-        return self.base[i, j]**self.exp
+        from sympy.core.compatibility import Iterable
+        base = self.base
+        exp = self.exp
+
+        if base.is_Matrix:
+            if exp.is_Matrix:
+                return self.base[i, j]**self.exp[i, j]
+            elif not isinstance(exp, Iterable):
```

Comment:
What happens if exp is Iterable? What does that correspond to?

Reply:
I think Iterable check can detect scalar exponent.

Reply:
My question is what us the third possibility in this if/elif chain and what happens in that case?

Reply:
Okay, I see some exception may be added here.